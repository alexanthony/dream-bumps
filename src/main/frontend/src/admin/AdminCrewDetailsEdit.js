import React, { Component, PropTypes } from 'react';
import {connect} from 'react-redux';
import {Field, reduxForm} from 'redux-form';
import {Card, CardTitle, CardText} from 'material-ui/Card';
import {TextField, SelectField} from 'redux-form-material-ui';
import RaisedButton from 'material-ui/RaisedButton';
import MenuItem from 'material-ui/MenuItem';

import {getCrew} from '../rootReducer';
import {updateCrew} from '../crews/actions';
/**
 * AdminCrewDetailsEdit
 */
class AdminCrewDetailsEdit extends Component {
  render() {
    return (
      <Card>
        <CardTitle title={`Edit Crew - ${this.props.crewId}`}/>
        <CardText>
          <form onSubmit={this.props.handleSubmit}>
            <Field
              name="name"
              floatingLabelText="Name"
              component={TextField}
              fullWidth
            />
            <Field
              name="sex"
              component={SelectField}
              floatingLabelText="Sex"
              fullWidth
            >
              <MenuItem value="male" primaryText="Male" />
              <MenuItem value="female" primaryText="Female" />
            </Field>
            <RaisedButton primary type="submit" label="Save"/>
          </form>
        </CardText>
      </Card>
    );
  }
}

AdminCrewDetailsEdit.propTypes = {
}

const mapStateToProps = (state, {crewId}) => ({
  initialValues: getCrew(state)(crewId)
});

export default connect(mapStateToProps, {onSubmit: updateCrew})(reduxForm({form: 'edit-crew-details', enableReinitialize: true})(AdminCrewDetailsEdit));
