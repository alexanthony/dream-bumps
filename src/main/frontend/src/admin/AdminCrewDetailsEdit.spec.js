import React from 'react';
import { shallow } from 'enzyme';
import { AdminCrewDetailsEdit } from './AdminCrewDetailsEdit';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const submit = jest.fn();
  const component = shallow(
    <AdminCrewDetailsEdit
      crewId="15"
      initialValues={{name: "Crew Name", sex: "male"}}
      handleSubmit={submit}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});
