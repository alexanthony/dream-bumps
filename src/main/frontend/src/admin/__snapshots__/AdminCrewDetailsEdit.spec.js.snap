exports[`test Renders properly 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}>
  <CardTitle
    title="Edit Crew - 15" />
  <CardText>
    <form
      onSubmit={[Function]}>
      <Field
        component={[Function]}
        floatingLabelText="Name"
        fullWidth={true}
        name="name" />
      <Field
        component={[Function]}
        floatingLabelText="Sex"
        fullWidth={true}
        name="sex">
        <MenuItem
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          primaryText="Male"
          value="male" />
        <MenuItem
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          primaryText="Female"
          value="female" />
      </Field>
      <RaisedButton
        disabled={false}
        fullWidth={false}
        label="Save"
        labelPosition="after"
        primary={true}
        secondary={false}
        type="submit" />
    </form>
  </CardText>
</Card>
`;
